### PR TITLE
add sample for hot reloading typescript lambdas

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

File: lambda-hot-reloading/typescript/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+```sh
+{"statusCode":200,"body":"{\"message\":\"Hello World!\"}"}
+```
+
+## Changing the Lambda Function
+
+The Lambda function is now mounted as a file in the executing container, hence any change that we save on the file will be there in an instant.
+
+Change the `Hello World!` message to `Hello LocalStack!` and run `npm run build`. Trigger the Lambda once again. You will see the following in the `output.txt` file:
```
Comment (dfangl) [inline lambda-hot-reloading/typescript/README.md:96]: If we run --watch, we should not need to run `npm run build` again here. Also, the process might stay up while watching for changes (?) I haven't tried this yet to be fair, but it should be the case, I guess?

File: lambda-hot-reloading/typescript/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+- Set the S3 key to the path of the directory your lambda function resides in. The handler is then referenced by the filename of your lambda code and the function in that code that needs to be invoked.
+
+Create the Lambda Function using the `awslocal` CLI:
+
+```sh
+awslocal lambda create-function \
+    --function-name hello-world \
+    --runtime "nodejs16.x" \
+    --role arn:aws:iam::123456789012:role/lambda-ex \
+    --code S3Bucket="hot-reload",S3Key="$(PWD)/dist" \
```
Comment (dfangl) [inline lambda-hot-reloading/typescript/README.md:65]: Not sure for mac, but for linux, `$(PWD)` is not valid. You either do `$(pwd)` (in essence executing the command `pwd` in a subshell, most likely it is a shell-builitin, and getting it's output in there) or you do `${PWD}`, which will get the value of the environment variable `PWD`, which is more or less what the shell builtin `pwd` does as well per default (in contrast to the binary `pwd`). There are edge cases for symbolic links though.

Long story short, I think we should either change to `${PWD}` or `$(pwd)`

File: lambda-hot-reloading/typescript/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+Add a build script to your `package.json` file:
+
+```json
+"scripts": {
+    "build": "esbuild index.ts --bundle --minify --sourcemap --platform=node --target=es2020 --outfile=dist/index.js --watch"
+}
+```
+
+The build script will use `esbuild` to bundle and minify the TypeScript code into a single JavaScript file, which will be placed in the `dist` folder.
```
Comment (dfangl) [inline lambda-hot-reloading/typescript/README.md:43]: If --watch really works properly here, it might be good to amend that description to make it clear it will watch for changes.

**Review by dfangl — CHANGES_REQUESTED** [review]
Just some minor issues, mostly for clarification of --watch, but one command seems to be invalid (at least on linux/bash).
